app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -0,0 +1,193 @@
+                errorView.text = errorStatus.first
+                errorView.isVisible = errorStatus.second
+            }
+            layerNameInput.addTextChangedListener {
+                newLayerName = it.toString().trim()
+                onEditText?.invoke(newLayerName)
+                val errorStatus = checkForErrors(context, layerList, newLayerName)
+                errorView.text = errorStatus.first
+                errorView.isVisible = errorStatus.second
+            }
+            setPositiveButton(context.getString(R.string.addLayerPositiveButton)) { _, _ ->
+                if (errorView.isVisible) {
+                    onFailure?.invoke(errorView.text as String)
+                } else {
+                    when (newLayerName) {
+                        NAME_EMPTY, currentHint -> {
+                            newLayerName = currentHint
+                            onNameLayer?.invoke(newLayerName)
+                        }
+                        else -> {
+                            onNameLayer?.invoke(newLayerName)
+                        }
+                    }
+                }
+                onDialogComplete?.invoke()
+            }
+            setNegativeButton(context.getString(R.string.addLayerNegativeButton)) { _, _ ->
+                onDialogComplete?.invoke()
+            }
+            setOnCancelListener {
+                onDialogComplete?.invoke()
+            }
+        }
+        dialog.show()
+    }
+
+    fun layerListBottomSheet(
+        context: Context,
+        layerList: MutableList<String>,
+        /**
+         * Opens a Dialog to Delete an object in the list.
+         */
+        onSelectDelete: ((Int) -> Unit)?,
+        /**
+         * Opens a Dialog to Edit an object in the list.
+         */
+        onSelectEdit: ((Int?) -> Unit)?,
+        /**
+         * Called when the Dialog Completes through any listener.
+         * This will reset all values to null in preparation for the next opening of the Dialog.
+         */
+        onDialogComplete: (() -> Unit)?,
+    ): ((Int?, String) -> Unit) {
+        val sheet = BottomSheetDialog(context)
+        val inflater = LayoutInflater.from(context)
+        val view = inflater.inflate(R.layout.recycler_layer_list_layout, null)
+        val data = mutableListOf<AdapterViewModel>()
+        view.findViewById<Button>(R.id.closeLayerListView).setOnClickListener {
+            onDialogComplete?.invoke()
+            sheet.dismiss()
+        }
+        (layerList.size - 1 downTo 0).forEach {
+            val layerListNameAtIndex = layerList[it]
+            data.add(
+                data.size, LayerViewModel(layerName = layerListNameAtIndex,
+                    onDeleteLayer = { onSelectDelete?.invoke(it) },
+                    onEditName = { onSelectEdit?.invoke(it) })
+            )
+        }
+        val layerListRecycler = view.findViewById<RecyclerView>(R.id.listOfLayers)
```

Comment:
Since this is a `RecyclerView` and someone might confuse this at first glance for a `Recycler` `Adapter`, it may be nice just to put `View` at the end of this name